srcs/incs/RequestConfig.hpp

```diff
@@ -0,0 +1,55 @@
+#ifndef SRCS_INCS_REQUESTCONFIG_HPP_
+#define SRCS_INCS_REQUESTCONFIG_HPP_
+#include <string>
+#include <vector>
+#include <algorithm>
+#include <ServerConfig.hpp>
+
+// All config related to a request in a Sever
+
+class RequestConfig {
+	public:
+		RequestConfig(const ServerConfig &server_config,
+						const std::string &request_path);
+		RequestConfig();
+		RequestConfig(const RequestConfig &);
+		std::string	getPath() const;
+		bool limits(std::string method) const;
+		std::string	getRoot() const;
+		uint32_t	getClientMaxBodySize() const;
+		bool	hasAutoindex() const;
+		std::string	getIndex() const;
+		std::string	getUploadStore() const;
+		uint16_t	getReturnStatus() const;
+		std::string	getReturnUrl() const;
+		bool	hasReturnUrl() const;
+		bool	hasCGI(std::string extension) const;
+		std::string	getErrorPagePath(std::size_t errCode) const;
+		CommonConfig::BinaryHandlerPath getCGIBin(std::string extension) const;
+
+		// temporal
+		const CommonConfig	&getCommonConfig() const;
+
+	private:
+		RequestConfig &	operator=(const RequestConfig &);
```

Comment:
Why is this here? Is it implemented?

Reply:
A lot of work in these PRs is about moving things aroung and cleaning at the end. This may be one of those left uncleaned. Thanks!